### PR TITLE
winch: implement div for aarch64

### PR DESCRIPTION
**Pull request by MarinPostma**
Implement integer division for arch64

### CODE REVIEW

**Comment by cfallin** [comment]
passing review over to @saulecabrera if you don't mind!

File: winch/codegen/src/isa/aarch64/asm.rs [hunk 1]
```diff
@@ -397,6 +399,80 @@ impl Assembler {
...
+            });
+
+            // Finally, trap if the previous operation overflowed
+            self.emit(Inst::TrapIf {
+                kind: CondBrKind::Cond(Cond::Vs),
+                trap_code: TrapCode::INTEGER_OVERFLOW,
+            })
+        }
+
+        // cranelif-codegen doesn't support emitting u/sdiv for anything but I64,
```
Comment (saulecabrera) [inline winch/codegen/src/isa/aarch64/asm.rs:443]:
```suggestion
        // `cranelift-codegen` doesn't support emitting u/sdiv for anything but I64,
```

Comment (saulecabrera) [reply winch/codegen/src/isa/aarch64/asm.rs:443]: Would you mind linking to this issue from this comment? https://github.com/bytecodealliance/wasmtime/issues/9766

File: winch/codegen/src/isa/aarch64/asm.rs [hunk 2]
```diff
@@ -397,6 +399,80 @@ impl Assembler {
...
+                to_bits: 64,
+            });
+        }
+
+        let alu_op = match kind {
+            DivKind::Signed => ALUOp::SDiv,
+            DivKind::Unsigned => ALUOp::UDiv,
+        };
+
+        self.emit(Inst::AluRRR {
```
Comment (saulecabrera) [inline winch/codegen/src/isa/aarch64/asm.rs:467]: Here you can use the `emit_alu_rrr` helper defined in this file.

**Review by saulecabrera — COMMENTED** [review]
Looks reasonable to me; a couple of nitpicks inline.

**Review by saulecabrera — APPROVED** [review]

**Pull request merged**